### PR TITLE
Add support to check dnsConfig options

### PR DESCRIPTION
**Pull request by weixiongny**
It is for feature request https://github.com/stackrox/kube-linter/issues/354
Add support to check dnsConfig options.
Thanks @trumant for the initial discussion.

### CODE REVIEW

File: pkg/builtinchecks/yamls/dnsconfig-options.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+name: "dnsconfig-options"
```
Comment (trumant) [inline pkg/builtinchecks/yamls/dnsconfig-options.yaml:1]: Is this implementation too `ndots` specific? In other words, does the solution as written provide a mechanism to check other aspects of `dnsConfig` like `nameservers` and `searches`?

File: docs/generated/checks.md [hunk 2]
```diff
@@ -118,6 +118,22 @@ serviceAccount: ^(|default)$
...
 **Template**: [deprecated-service-account-field](generated/templates.md#deprecated-service-account-field)
+## dnsconfig-options
+
+**Enabled by default**: No
+
+**Description**: Alert on deployments that have no specified dnsconfig options
+
+**Remediation**: Specify dnsconfig options in your pod specification to ensure the expected dns setting on the pod. Refer to https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-dns-config for details.
+
+**Template**: [dnsconfig-options](generated/templates.md#dnsconig-options)
```
Comment (mdb) [inline docs/generated/checks.md:129]:
```suggestion
**Template**: [dnsconfig-options](generated/templates.md#dnsconfig-options)
```

File: pkg/builtinchecks/yamls/dnsconfig-options.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+description: "Alert on deployments that have no specified dnsconfig options"
```
Comment (mdb) [inline pkg/builtinchecks/yamls/dnsconfig-options.yaml:2]: Should `dnsConfig` be camelcased to match how it appears in the spec YAML?

```suggestion
description: "Alert on deployments that have no specified dnsConfig options"
```

File: pkg/builtinchecks/yamls/dnsconfig-options.yaml (hunk 1 above)
Comment (weixiongny) [reply pkg/builtinchecks/yamls/dnsconfig-options.yaml:1]: Current implementation supports matching in `options` field, such as `ndots`, `timeout`, `attempts`, etc. A full list [here under options section](https://linux.die.net/man/5/resolv.conf). `nameservers` and `searches` are at the same level as `options`, so not supported currently.

**Review by janisz — APPROVED** [review]

**Review by viswajithiii — APPROVED** [review]
Thanks @weixiongny!

**Comment by weixiongny** [comment]
Hi @viswajithiii @janisz , any chance for merging this pr? Thanks!

**Pull request merged**